src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs

```diff
@@ -170,5 +179,81 @@ protected void RemoveDocument(HostProject hostProject, string filePath)
                 _currentDocuments.Remove(filePath);
             }
         }
+
+        // Internal for testing
+        internal static bool TryGetIntermediateOutputPath(
```

Comment:
This is a pretty direct copy from the VSWindows variant (same with the fallback path)